### PR TITLE
build (deps): update GOLANGCI_VERSION

### PR DESCRIPTION
**Pull request by gssbzn**
I wanted to do this staged but the whole set up of the linter needed some improvements here which made me bring a bunch of changes I was not planning on 
This should address a couple of things
- We now use golintci action instead of us installing it manually in the CI
- Some linting fixes now that we have some extra rules
- Some makefile fixes based on me reading over it to improve linting

### CODE REVIEW

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -1,21 +1,43 @@
...
-  tests:
-    name: Pull Request
-    runs-on: ubuntu-20.04
+  golangci-lint:
+    runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-
-      - uses: actions/setup-go@v2
+      - name: lint
```
Comment (gssbzn) [inline .github/workflows/pr.yml:15]: golang ci has their own action, let's use that

File: Makefile [hunk 2]
```diff
@@ -28,20 +31,17 @@ fix-lint: ## Fix linting errors
...
 	@echo "==> Running tests"
-	@go test ./...
+	@go test -race -cover -count=1 -coverprofile ${COVERAGE} ./...
+
+.PHONY: check
+check: test fix-lint
 
 .PHONY: all
-all: fmt lint build test ## Run all targets
+all: fmt test lint ## Run all targets
```
Comment (gssbzn) [inline Makefile:44]: we should fallow a red -> green -> refactor approach, so tests first, lint latter

File: Makefile [hunk 3]
```diff
@@ -28,20 +31,17 @@ fix-lint: ## Fix linting errors
 	@echo "==> Fixing lint errors"
 	@./bin/golangci-lint run --fix
 
-.PHONY: build
-build: ## Build the library
```
Comment (gssbzn) [inline Makefile:32]: this is a lib, no need to build

File: digest_test.go [hunk 4]
```diff
@@ -37,41 +37,47 @@ func TestH(t *testing.T) {
...
 			"expected_r1": "3ba6cd94661c5ef34598040c868f13b8775df29109986be50ad35ae537dd3aa4",
 			"expected_r2": "9a3fdae9a622fe8de177c24fa9c070f2b181ec85e15dcbdc32e10c82ad450b04",
-			"expected_r3": "5abdd07184ba512a22c53f41470e5eea7dcaa3a93a59b630c13dfe0a5dc6e38b",
+			"expected_r3": "705e2bcd4503344546454cc767a5e3107dc09f86dc898d840fff18b257f0b62f",
 		},
 	}
 
 	for testType, values := range testCases {
-		t.Run(testType, func(t *testing.T) {
+		r1 := values["r1"]
```
Comment (gssbzn) [inline digest_test.go:52]: I enabled a linter for this, loops + function in go can be tricky with pointers as they'll keep the reference to the last element of the loop, this also exposed a fail in the tests for the same reason

File: .github/workflows/pr.yml [hunk 5]
```diff
@@ -1,21 +1,45 @@
...
+          - 1.14
+          - 1.15
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.golang }}
+      - name: Cache Dependencies
```
Comment (MihaiBojin) [inline .github/workflows/pr.yml:37]: nice!

**Review by MihaiBojin — APPROVED** [review]

**Pull request merged**